src/cargo/sources/git/known_hosts.rs

```diff
@@ -316,42 +391,75 @@ fn check_ssh_known_hosts_loaded(
             }
             continue;
         }
-        if key_matches {
-            return Ok(());
+        match known_host.line_type {
+            KnownHostLineType::Key => {
+                if key_matches {
+                    accepted_known_host_found = true;
+                }
+
+                // The host and key type matched, but the key itself did not.
+                // This indicates the key has changed.
+                // This is only reported as an error if no subsequent lines have a
+                // correct key.
+                latent_errors.push(KnownHostError::HostKeyHasChanged {
+                    hostname: host.to_string(),
+                    key_type: remote_key_type,
+                    old_known_host: known_host.clone(),
+                    remote_host_key: remote_host_key_encoded.clone(),
+                    remote_fingerprint: remote_fingerprint.clone(),
+                });
```

Comment:
I think the `latent_errors` part should be in an else block.  Otherwise a matching entry will always add a latent error. Although that won't cause any harm as it is written now, I think it would be good to avoid adding an incorrect error object.

Reply:
You're right. I'm sorry about that, I made that last change without thinking enough it seems. I've fixed this as suggested now.